### PR TITLE
Fix goroutine leaks in grpc integration tests

### PR DESCRIPTION
**Pull request by WillSewell**
## Which problem is this PR solving?
- Solves part of https://github.com/jaegertracing/jaeger/issues/5006

## Description of the changes
- The key issue was that the grpc client was only being killed in a `runtime.SetFinalizer` - i.e. when it is GCed. I think in the tests this was not shutting down before goleak had decided that the goroutine had leaked.
- This change switches to a more conventional approach of killing the client as part of the Close method and then ensuring this is called in each of the tests.

## How was this change tested?
- While this change does not include a call to activate goleak, it _was_ tested with this. The PR does not include goleak detection because there are still other violations related to elasticsearch in this package.

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by yurishkuro** [comment]
I merged a large PR that affected some of the test files. There are now merge conflicts (which would've been more difficult to resolve in that large PR than here, so I picked this one to resolve conflicts).

**Comment by WillSewell** [comment]
No worries. The changes actually make this PR simpler.

One area that is pretty unpleasant about this change is the fact that that each subtest is calling `CleanUp` which in this case is actually _initializing_ for the following subtest. This means the final subtest pointlessly initializes. For now I've taken the approach of just explicitly closing this after each test.

I think a nicer approach would be to introduce separate a `Initialize` callback (separate from `CleanUp`) but this is a larger change because I think it should be done across all integration tests.

**Comment by yurishkuro** [comment]
I also don't like how CleanUp functions often call initialize(), this was not at all how it was meant to work, the cleanup functions were intended to remove data from the DB, not to reset the connection completely. 

I am not opposed to larger refactoring (may not be as large as it seems) if we can make the workflow more robust and more intuitive.

I don't even remember why Refresh callback was needed - nothing implements it except for ES, and in case of ES the reason seems to be to ensure that the data is queryable after write once Refresh returns, which actually doesn't seem to work reliably anyway and we already have most of the assertions on reading data executing within waitForCondition, which makes synchronous flushing irrelevant.

File: plugin/storage/integration/grpc_test.go [hunk 1]
```diff
@@ -109,6 +113,7 @@ func TestGRPCStorage(t *testing.T) {
 	}
 	s.initialize(t)
 	s.RunAll(t)
+	s.close(t)
```
Comment (yurishkuro) [inline plugin/storage/integration/grpc_test.go:116]: nit: should really be called before RunAll via defer()

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**